pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,408 @@
+// host platforms and Linux-based containers.
+type LinuxPodSandboxConfig struct {
+	// CgroupParent is the parent cgroup of the sandbox. The cgroupfs style
+	// syntax will be used, but the container runtime can convert it to systemd
+	// semantices if needed.
+	CgroupParent string
+	// NamespaceOptions contains configurations for the sandbox's namespaces.
+	// This will be used only if the PodSandbox uses namespace for isolation.
+	NamespaceOptions NamespaceOptions
+}
+
+// NamespaceOptions provides options for Linux namespaces.
+type NamespaceOptions struct {
+	// HostNetwork uses the host's network namespace.
+	HostNetwork bool
+	// HostPID uses the host's pid namesapce.
+	HostPID bool
+	// HostIPC uses the host's ipc namespace.
+	HostIPC bool
+}
+
+// DNSOptions specifies the DNS servers and search domains.
+type DNSOptions struct {
+	// Servers is a list of DNS servers of the cluster.
+	Servers []string
+	// Searches is a list of DNS search domains of the cluster.
+	Searches []string
+}
+
+type PodSandboxState string
+
+const (
+	// PodSandboxReady means the sandbox is functioning properly.
+	PodSandboxReady PodSandboxState = "Ready"
+	// PodSandboxInNotReady means the sandbox is not functioning properly.
+	PodSandboxNotReady PodSandboxState = "NotReady"
+)
+
+// PodSandboxFilter is used to filter a list of PodSandboxes.
+type PodSandboxFilter struct {
+	// Name of the sandbox.
+	Name *string
+	// ID of the sandbox.
+	ID *PodSandboxID
+	// State of the sandbox.
+	State *PodSandboxState
+	// LabelSelector to select matches.
+	// Only api.MatchLabels is supported for now and the requirements
+	// are ANDed. MatchExpressions is not supported yet.
+	LabelSelector unversioned.LabelSelector
+}
+
+// PodSandboxListItem contains minimal information about a sandbox.
+type PodSandboxListItem struct {
+	ID    PodSandboxID
+	State PodSandboxState
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+}
+
+// PodSandboxStatus contains the status of the PodSandbox.
+type PodSandboxStatus struct {
+	// ID of the sandbox.
+	ID PodSandboxID
+	// State of the sandbox.
+	State PodSandboxState
+	// Network contains network status if network is handled by the runtime.
+	Network *PodSandboxNetworkStatus
+	// Status specific to a Linux sandbox.
+	Linux *LinuxPodSandboxStatus
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map.
+	Annotations map[string]string
+}
+
+// PodSandboxNetworkStatus is the status of the network for a PodSandbox.
+type PodSandboxNetworkStatus struct {
+	IP string
+}
+
+// Namespaces contains paths to the namespaces.
+type Namespaces struct {
+	// Network is the path to the network namespace.
+	Network string
+}
+
+// LinuxSandBoxStatus contains status specific to Linux sandboxes.
+type LinuxPodSandboxStatus struct {
```

Comment:
What in kubelet itself or the k8s API is expected to use the netns path?  I think that's really an implementation detail of the runtime itself and should be isolated in the runtime code and network plugins.  I don't see kubelet or API consumers needing the netns path at all.

Obviously the network plugins care, but since they are called by the runtime itself I think we should enhance the network plugin API to allow passing namespace/VM-id/whatever into the plugin at SetUpPod/TearDownPod/GetPodNetworkStatus time.  But I don't think this needs to be part of the kubelet <-> runtime API at all?  Kubelet shouldn't care.